reframe/core/schedulers/slurm.py

```diff
@@ -421,6 +434,13 @@ def __hash__(self):
     def is_available(self):
         return self._state == 'IDLE'
 
+    def is_down(self):
+        return any(
+            self.in_state(s) for s in ['DOWN', 'DRAIN', 'MAINT', 'NO_RESPOND'])
+
+    def in_state(self, state):
+        return state in self._state
+
```

Comment:
I think it would be better if `self._state` was actually a set holding all the states a Slurm node is in. Then `is_down` could be rewritten more efficiently as follows:

```python
def is_down(self):
    return bool({'DOWN', 'DRAIN', 'MAINT', 'NO_RESPOND'} & self._states)
```